thrust/thrust/system/omp/detail/scan.h

```diff
@@ -106,7 +106,25 @@ OutputIterator scan_impl(
 
     if (start < n)
     {
-      block_sums[tid] = ::cuda::std::reduce(first + start, first + end, accum_t{}, wrapped_binary_op);
+      if constexpr (has_init)
+      {
+        if (tid == 0)
+        {
+          block_sums[tid] = ::cuda::std::reduce(first + start, first + end, init, wrapped_binary_op);
+        }
+        else
+        {
+          block_sums[tid] = ::cuda::std::reduce(first + start, first + end, accum_t{}, wrapped_binary_op);
+        }
+      }
+      else
+      {
+        // No init value: reduce uses first element as init
+        if (end > start)
+        {
+          block_sums[tid] = ::cuda::std::reduce(first + start + 1, first + end, *(first + start), wrapped_binary_op);
+        }
+      }
```

Comment:
You want to always do the `else` branch.  The true branch is incorrect.  The initial value doesn't get incorporated into the result until step 2.  And you still have `accum_t{}` in there, which is not correct.  Get rid of the `if` statement entirely and just replace
```c++
block_sums[tid] = ::cuda::std::reduce(first + start, first + end, accum_t{}, wrapped_binary_op);
```
with
```c++
block_sums[tid] = ::cuda::std::reduce(first + start + 1, first + end, *(first + start), wrapped_binary_op);
```